### PR TITLE
Add credMethods to StoredAccount.

### PR DESCRIPTION
**Pull request by aforge**
Following the logic of https://github.com/tinode/tindroid/commit/5e13af04bcb7fa62c22bbd9c423a278c88e14004

### CODE REVIEW

File: TinodeSDK/Tinode.swift [hunk 1]
```diff
@@ -768,18 +768,31 @@ public class Tinode {
...
         if ctrl.code < 300 {
-            isConnectionAuthenticated = true
-            if let t = authToken, !autoLogin {
-                setAutoLoginWithToken(token: t)
+            store?.myUid = newUid
+            // Load topics if not yet loaded.
+            loadTopics()
+        } else {
+            var meth: [String]? = nil
+            if let params = ctrl.params {
```
Comment (or-else) [inline TinodeSDK/Tinode.swift]: Why not just 
```
var meth = ctrl.params?.getStringArray(for: "cred")
```

File: TinodeSDK/Tinode.swift [hunk 2]
```diff
@@ -768,18 +768,31 @@ public class Tinode {
...
+                        meth!.append(m)
+                    }
+                }
             }
-            listener?.onLogin(code: ctrl.code, text: ctrl.text)
+            store?.setMyUid(uid: newUid!, credMethods: meth)
+        }
+        isConnectionAuthenticated = true
+        if let t = authToken, !autoLogin {
+            setAutoLoginWithToken(token: t)
```
Comment (or-else) [inline TinodeSDK/Tinode.swift]: Interesting. This is different from Android. On Android the app calls setAutoLoginWithToken. Here the SDK calls it, the app loses ability to control setAutoLoginWithToken. Maybe then it should be made private?

If `autoLogin` is true the token is not saved. Does not seem logical to me.

File: Tinodios/db/AccountDb.swift [hunk 3]
```diff
@@ -56,39 +61,47 @@ public class AccountDb {
...
+                    result = StoredAccount(id: newId, uid: uid, credMethods: meth)
                 }
                 if result!.id < 0 {
                     result = nil
+                } else {
+                    result?.credMethods = meth
                 }
             }
         } catch {
+            print("failed to add account '\(error)'")
```
Comment (or-else) [inline Tinodios/db/AccountDb.swift]: `Log.debug....` etc?

File: TinodeSDK/Tinode.swift (hunk 2 above)
Comment (aforge) [reply TinodeSDK/Tinode.swift]: Ah, okay. This one was copied over from an earlier Android version.
Yes, I agree it should be called from the app.

**Pull request merged**